lib/Report/ProofOfPlay.php

```diff
@@ -1228,4 +1251,115 @@ private function groupByTagType(string $tagType) : string
                          `lktagdisplaygroup` AS taglink ON taglink.displaygroupId = displaydg.displaygroupId',
         };
     }
+
+    /**
+     * Group by display group in MongoDB
+     * @param array $rows
+     * @param array $displayIds
+     * @return array
+     * @throws NotFoundException
+     */
+    private function groupByDisplayGroupMongoDb(array $rows, array $displayIds = []) : array
+    {
+        $data = [];
+
+        // Get the display groups
+        foreach ($rows as $row) {
+            foreach ($this->displayGroupFactory->query(null, ['displayId' => $row['displayId']]) as $dg) {
+
+                // Do we have a filter?
+                if (!$displayIds || in_array($dg->displayGroupId, $displayIds)) {
```

Comment:
We're checking here whether the displayGroupId is in an array of displayIds (for filtering purposes), but will that ever be true?